cpp/src/arrow/engine/substrait/extension_set.h

```diff
@@ -66,12 +66,12 @@ constexpr const char* kSubstraitAggregateGenericFunctionsUri =
     "https://github.com/substrait-io/substrait/blob/main/extensions/"
     "functions_aggregate_generic.yaml";
 
-struct Id {
+struct ARROW_ENGINE_EXPORT Id {
```

Comment:
What are these changes for?

Reply:
They are Windows specific and declare that a class/method is usable outside the DLL.  It is basically used to mark any method or class that might be used by those linking to your DLL (e.g. your public API).  My guess was that since these types will be useful for those writing extensions they need to be exported.  I didn't check to see if the export was absolutely necessary but it makes sense to me that this be part of the public API.

Reply:
I feel it's orthogonal to the purpose of this PR and therefore probably should be a separate PR to avoid confusion?

Reply:
I vaguely recall I had a reason for it in this PR. I can try to remove and see.